### ISSUE TITLE
[BUG] vsphere cluster creation fails when using vapp network protocol profiles

### ISSUE THREAD

**Issue by philipp1992** [rancher/dashboard#12271]
Using the latest rancher version, when creating a new vsphere cluster, it fails upon cloning the vm template to a virtual machine with the following error:

![image](https://user-images.githubusercontent.com/11994884/183889750-7b425b7a-4e6c-45a4-924a-d208124eeccb.png)

settings:
![image](https://user-images.githubusercontent.com/11994884/183889812-846242b1-70c7-481a-a000-31b798954458.png)

Network protocol profile is configured as following:

![image](https://user-images.githubusercontent.com/11994884/183890011-b448d532-8323-49d2-aa80-fad5761866e9.png)

have also tried with empty domain / search but same issue

**Comment by izaac** [comment]
The vSphere test environment is offline I'll come back to this issue once that's sorted out

**Comment by izaac** [comment]
Validated on Rancher version: `v2.10-73aadc325c6789878e44579da1cba61749f2ed70-head` - HA k3s 1.29.
UI Version: `release-2.10 413828f`
vSphere version: `8`

This is now validated working on Rancher, with a properly configured network protocol profile in the Datacenter I was able to use the option: `Use vApp to configure networks with network protocol profiles` with the network associated with it in the network dropdown.

No errors were observed. I validated the negative scenario too, without a profile and a badly configured profile.
